File: day1-1.py
def get_calibration_value(string: list[str]):
    calibration_value_list = list()

    for char in string:
        if char.isdecimal():
            calibration_value_list.append(char)
            break

    for char in reversed(string):
        if char.isdecimal():
            calibration_value_list.append(char)
            break

    return int("".join(calibration_value_list))


with open("day1-1.input", "r") as f:
    data = list(map(list, f.readlines()))

    print(sum(map(get_calibration_value, data)))

File: kotlin/src/main/kotlin/day1/Day1_1.kt
package day1

import java.io.File

fun readFile(): List<String> {
    val fileName = "day1-1.input"
    val file = File("./kotlin/src/main/kotlin/day1/$fileName")
    val lines = file.readLines()
    return lines
}

fun getCalibrationValue(input: String): Int {
    val calibrationValueList = mutableListOf<Int>()

    for (char in input) {
        if (char.isDigit()) {
            calibrationValueList.add(Integer.parseInt(char.toString()))
            break
        }
    }

    for (char in input.reversed()) {
        if (char.isDigit()) {
            calibrationValueList.add(Integer.parseInt(char.toString()))
            break
        }
    }

    return calibrationValueList.joinToString("").toInt()
}

fun main() {
    val inputs = readFile()

    println(inputs.sumOf { getCalibrationValue(it) })
}


